src/amoc_metrics.erl

```diff
@@ -1,122 +1,107 @@
 -module(amoc_metrics).
 
--export([start/0, init/2, user_size/0]).
--export([update_time/2, update_counter/2, update_counter/1, update_gauge/2]).
+-behaviour(prometheus_collector).
+
+-export([start/0, start_predefined_metrics/1, init/2]).
+-export([update_counter/1, update_counter/2, update_gauge/2, update_time/2]).
+-export([deregister_cleanup/1, collect_mf/2]).
 
 -include_lib("kernel/include/logger.hrl").
 
 -type simple_name() :: atom() | [atom()].
 -type name() :: simple_name() | {strict, simple_name()}.
--type type() :: counters | times | gauge.
+-type type() :: counters | gauge | times | histogram | summary.
 
 -export_type([name/0]).
 
--define(AMOC_DEFAULT_METRICS_REPORTER, exometer_report_graphite).
--define(AMOC_METRICS_REPORTING_INTERVAL, timer:seconds(10)).
+%% ===================================================================
+%% API
+%% ===================================================================
 
--spec start() -> any().
+-spec start() -> boolean().
 start() ->
-    maybe_add_reporter(),
-    subsribe_default_metrics(),
-    maybe_init_predefined_metrics().
-
--spec init(type(), name()) -> ok.
-init(Type, Name) ->
-    ExName = make_name(Type, Name),
-    ExType = exometer_metric_type(Type),
-    create_metric_and_maybe_subscribe(ExName, ExType).
+    maybe_add_exporter().
 
--spec user_size() -> [{size, non_neg_integer()}].
-user_size() ->
-    [{size, amoc_users_sup:count_no_of_users()}].
+-spec start_predefined_metrics(atom()) -> any().
+start_predefined_metrics(App) ->
+    Preconfigured = application:get_env(App, predefined_metrics, []),
+    [init(Type, Name) || {Type, Name} <- lists:flatten(Preconfigured)].
 
--spec update_time(name(), integer()) -> ok.
-update_time(Name, Value) ->
-    ExName = make_name(times, Name),
-    exometer:update(ExName, Value).
+-spec init(type(), name()) -> ok.
+init(counters, Name) ->
+    prometheus_counter:new([{name, Name}]);
+init(gauge, Name) ->
+    prometheus_gauge:new([{name, Name}]);
+init(summary, Name) ->
+    prometheus_summary:new([{name, Name}]);
+init(Type, Name) when histogram =:= Type; times =:= Type ->
+    prometheus_histogram:new([{name, Name}, {buckets, histogram_buckets()}]).
 
 -spec update_counter(name()) -> ok.
 update_counter(Name) ->
-    ExName = make_name(counters, Name),
-    exometer:update(ExName, 1).
+    prometheus_counter:inc(Name).
 
 -spec update_counter(name(), integer()) -> ok.
 update_counter(Name, Value) ->
-    ExName = make_name(counters, Name),
-    exometer:update(ExName, Value).
+    prometheus_counter:inc(Name, Value).
 
 -spec update_gauge(name(), integer()) -> ok.
 update_gauge(Name, Value) ->
-    ExName = make_name(gauge, Name),
-    exometer:update(ExName, Value).
+    prometheus_gauge:inc(Name, Value).
```

Comment:
This works differently - before, the value would be replaced with the new one, and now it is incremented instead.
```suggestion
    prometheus_gauge:set(Name, Value).
```